app/src/androidTest/java/com/github/houseorganizer/houseorganizer/FirestoreShopItemUnitTest.java

```diff
@@ -0,0 +1,155 @@
+
+    @BeforeClass
+    public static void setUpTests() throws ExecutionException, InterruptedException {
+        FirebaseTestsHelper.startFirestoreEmulator();
+        FirebaseTestsHelper.startAuthEmulator();
+
+        db = FirebaseFirestore.getInstance();
+
+        FirebaseTestsHelper.setUpFirebase();
+
+        CollectionReference colRef = db.collection("shop_lists")
+                .document(FirebaseTestsHelper.TEST_SHOPLIST_NAME).collection("items");
+
+        List<FirestoreShopItem> list = FirestoreShopItem.retrieveShopItems(colRef);
+
+        testLocalItem = list.get(0);
+        testLocalItem.setPickedUp(false);
+        testLocalItem.setUnit(FirebaseTestsHelper.TEST_ITEM.getUnit());
+        testLocalItem.setQuantity(FirebaseTestsHelper.TEST_ITEM.getQuantity());
+        testLocalItem.changeName(FirebaseTestsHelper.TEST_ITEM.getName());
+    }
+
+    @Test
+    public void constructorCreatesItem(){
+        assertThat(testLocalItem.getName(), is("Egg"));
+    }
+
+    @Test
+    public void togglePickedUpPicksUpItem() throws ExecutionException, InterruptedException {
+        testLocalItem.togglePickedUp();
+        DocumentReference documentReference = testLocalItem.getItemDocRef();
+        Task<DocumentSnapshot> t = documentReference.get();
+        Tasks.await(t);
+        assertThat(t.getResult().get("pickedUp"), is(true));
+        testLocalItem.togglePickedUp();
+    }
+
+    @Test
+    public void changeNameUpdatesFirestore() throws ExecutionException, InterruptedException {
+        DocumentReference docRef = testLocalItem.getItemDocRef();
+        Task<DocumentSnapshot> t = docRef.get();
+        Tasks.await(t);
+        String oldName = (String)t.getResult().get("name");
+        testLocalItem.changeName("RandomNewName");
+        Task<DocumentSnapshot> t1 = docRef.get();
+        Tasks.await(t1);
+        String newName = (String)t1.getResult().get("name");
+        assertThat(oldName, is(not(newName)));
+        assertThat(newName, is("RandomNewName"));
+        testLocalItem.changeName(FirebaseTestsHelper.TEST_ITEM.getName());
+    }
+
+    @Test
+    public void setQuantityUpdatesFirestore() throws ExecutionException, InterruptedException {
+        DocumentReference docRef = testLocalItem.getItemDocRef();
+        Task<DocumentSnapshot> t = docRef.get();
+        Tasks.await(t);
+        long oldQ = (long)t.getResult().get("quantity");
+        testLocalItem.setQuantity(9);
+        Task<DocumentSnapshot> t1 = docRef.get();
+        Tasks.await(t1);
+        long newQ = (long)t1.getResult().get("quantity");
+        assertThat(oldQ, is(not(newQ)));
+        assertThat(newQ, is(9L));
+        testLocalItem.setQuantity(FirebaseTestsHelper.TEST_ITEM.getQuantity());
+    }
+
+    @Test
+    public void setUnitUpdatesFirestore() throws ExecutionException, InterruptedException {
+        DocumentReference docRef = testLocalItem.getItemDocRef();
+        Task<DocumentSnapshot> t = docRef.get();
+        Tasks.await(t);
```

Comment:
Those two lines are used a lot of times, you could extract them

Reply:
Probably yes I can